### PR TITLE
Fix validation conditions to prevent size calculation for countries

### LINKED ISSUE #311: disable validity checker for countries
With the exception of very minuscule countries, they cannot be selected.

There should be a distinction between countries and existing excerpts and the frontend-side validation disabled for countries.

### PR DESCRIPTION
**Pull request by wasabideveloper**
Fixes #311 the validation for countries: Bounds are not checked on selecting an existing excerpt or country.

To be reviewed by:
- [x] @hixi 
- [x] @das-g

### CODE REVIEW

File: osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/formValidation.js [hunk 1]
```diff
@@ -22,6 +22,7 @@
         };
 
         this.isFormValid = function() {
+            console.log(this.validity);
```
Comment (hixi) [inline osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/formValidation.js]: Is this `console.log` a remnant from development or does it serve a purpose?

File: osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/formValidation.js [hunk 2]
```diff
@@ -38,13 +39,15 @@
 
         this.validateExistingExcerptOrNew = function() {
             if(this.formModeSwitcher.value == "existing-excerpt") {
+            console.log('exist');
```
Comment (hixi) [inline osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/formValidation.js]: Is this `console.log` a remnant from development or does it serve a purpose?

File: osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/formValidation.js [hunk 3]
```diff
@@ -38,13 +39,15 @@
...
                 var selectValue = this.existingExcerptSelect.value;
-                if (!isNaN(parseFloat(selectValue)) && parseFloat(selectValue) > 0) {
+                if (selectValue && selectValue.length >= 1) {
                     this.validity['existingExcerptOrNew'] = true;
                 } else {
                     this.validity['existingExcerptOrNew'] = false;
                 }
             } else if(this.formModeSwitcher.value == "new-excerpt") {
+            console.log('new');
```
Comment (hixi) [inline osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/formValidation.js]: Is this `console.log` a remnant from development or does it serve a purpose?

File: osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/formValidation.js (hunk 2 above)
Comment (wasabideveloper) [inline osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/formValidation.js]: :heavy_check_mark:

File: osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/formValidation.js (hunk 3 above)
Comment (wasabideveloper) [inline osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/formValidation.js]: :heavy_check_mark:

File: osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/formValidation.js (hunk 1 above)
Comment (wasabideveloper) [inline osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/formValidation.js]: :heavy_check_mark:

File: osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/formValidation.js [hunk 4]
```diff
@@ -39,7 +39,7 @@
         this.validateExistingExcerptOrNew = function() {
             if(this.formModeSwitcher.value == "existing-excerpt") {
                 var selectValue = this.existingExcerptSelect.value;
-                if (!isNaN(parseFloat(selectValue)) && parseFloat(selectValue) > 0) {
+                if (selectValue && selectValue.length >= 1) {
                     this.validity['existingExcerptOrNew'] = true;
                 } else {
                     this.validity['existingExcerptOrNew'] = false;
```
Comment (das-g) [inline osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/formValidation.js]:
```js
                if (selectValue && selectValue.length >= 1) {
                    this.validity['existingExcerptOrNew'] = true;
                } else {
                    this.validity['existingExcerptOrNew'] = false;
                }
```
can be written more concisely as
```js
                this.validity['existingExcerptOrNew'] = selectValue && selectValue.length >= 1;
```

Comment (wasabideveloper) [inline osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/formValidation.js]: :heavy_check_mark:

File: osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/formValidation.js [hunk 5]
```diff
@@ -68,34 +60,40 @@
...
+                                var message = 'Excerpt {percent}too large!'.replace('{percent}', howMuchTooLarge);
+                                excerptBoundInputField.setCustomValidity(message);
+                                document.getElementById('excerpt-validation').textContent = message;
+                            }
+                        }.bind(this));
+                    }.bind(this)
+                );
+            } else {
+                this.validity['excerptBounds'] = true;
+                document.getElementById('excerpt-validation').textContent = '';
```
Comment (das-g) [inline osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/formValidation.js]: Not changed in this PR, but:

Can the ID of this element be changed to something more descriptive, e.g. `'excerpt-validation-message'` or `'excerpt-validation-result'`?

Also, some ID values contain an `'id_'` prefix, some don't. Is that intentional?

Comment (wasabideveloper) [inline osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/formValidation.js]: * If we refactor the forms, we won't need anymore this `excerpt-validation` container for error messages.
* Django form defaults will render this uncool `id_`. If we refactor the forms, we should try to get rid of this.

**Pull request merged**